cmd/func/main.go

```diff
@@ -1,14 +1,29 @@
 package main
 
 import (
+	"context"
 	"github.com/boson-project/func/cmd"
+	"os"
+	"os/signal"
+	"syscall"
 )
 
 // Statically-populated build metadata set
 // by `make build`.
 var date, vers, hash string
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I wonder how this would play with `kn` plugin system.
@rhuss would the plugin catch the signal, or `kn` would "steal" it?

Reply:
Well this code is probably not executed in plugin mode at all, am I right?

Reply:
Right, `kn` only calls the methods from the `Plugin` interface.

Reply:
Therefore it is important that you only call code in `main()` that is important for standalone execution, anything needed for initializing the plugin should go into the shared code